src/pages/index.js

```diff
@@ -32,30 +94,92 @@ const cardList = new Section(
   cardListSelector
 );
 
-cardList.renderItems(initialCards);
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__avatar"
+);
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    // take  user data from api and render to profile section
+    userInfo.setUserInfo(userData);
+    userInfo.setAvatar(userData);
+    // take initial cards from api and render to card section
+    cardList.renderItems(cards.reverse());
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const editAvatarModal = new PopupWithForm("#profile-edit-avatar", (link) => {
+  editAvatarModal.renderLoading(true);
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      userInfo.setAvatar(res);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarModal.renderLoading(false);
+    });
+});
+editAvatarModal.setEventListeners();
+
+document
+  .querySelector(".profile__avatar-overlay")
+  .addEventListener("click", () => {
+    editAvatarFormValidator.disableButton();
+    editAvatarModal.open();
+  });
 
 // Event Handlers
 function handleProfileEditSubmit(inputs) {
-  userInfo.setUserInfo(inputs);
+  profileEditModal.renderLoading(true);
+  api
+    .updateUserInfo(inputs)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      userInfo.setAvatar(res);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditModal.renderLoading(false);
+    });
 }
 
 function handleNewCardSubmit(inputs) {
-  renderCard(inputs);
+  addNewCardModal.renderLoading(true);
+  api
+    .addCard(inputs)
+    .then((res) => {
+      renderCard(res);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addNewCardModal.renderLoading(false);
+    });
 }
 
 const addNewCardModal = new PopupWithForm(
   "#add-new-card-modal",
   handleNewCardSubmit
 );
 addNewCardModal.setEventListeners();
-
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+addNewCardModal.renderLoading();
```

Comment:
Needs correction: There is no need to end the loading here, you should do that after the api request